### PR TITLE
feat(OnyxTable): support vertical/horizontal scrolling

### PR DESCRIPTION
**Pull request by BoppLi**
Relates to #840 

When the height/width of the table gets restricted, it scrolls.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 1]
```diff
@@ -165,14 +186,17 @@ const { densityClass } = useDensity(props);
...
-    th:hover::after,
+    // row hover styles
     tbody tr:hover td::before {
       background-color: var(--onyx-color-base-neutral-200);
     }
 
     // column hover styles
-    th:hover::after {
+    th:hover::before {
+      // dark mode:
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:196]: TODO adjust colors on dark/light mode

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx [hunk 2]
```diff
@@ -73,4 +73,50 @@ test.describe("Screenshot tests", () => {
...
   });
+
+  executeMatrixScreenshotTest({
+    name: "Table (scrolling)",
+    columns: ["default", "horizontal-scroll"],
+    rows: ["default", "vertical-scroll"],
+    // TODO: remove when contrast issues are fixed in https://github.com/SchwarzIT/onyx/issues/410
+    disabledAccessibilityRules: [
+      "color-contrast",
+      // TODO: FIX THIS!
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx:84]: TODO A11y issue with scroll content not being navigatible

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 3]
```diff
@@ -121,11 +156,18 @@ const { densityClass } = useDensity(props);
   @include layers.component() {
     @include define-borders();
 
+    --onyx-table-column-hover: #26628d30;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:159]:
```suggestion
    --onyx-table-column-hover: color-mix(
      in srgb,
      var(--onyx-color-base-neutral-300),
      transparent 50%
    );
```

We can use color-miz to transparentize CSS variables so we can use existing colors and also don't have to define special colors for dark mode :)

The values above are just examples, they don't look perfect

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 4]
```diff
@@ -18,29 +18,36 @@ const { densityClass } = useDensity(props);
...
-    :class="[
-      props.striped ? 'onyx-table--striped' : '',
-      props.grid ? 'onyx-table--grid' : '',
-      densityClass,
-    ]"
-  >
-    <slot></slot>
-  </table>
+  <div class="onyx-table-wrapper">
+    <div class="onyx-table-wrapper__scroll-container" tabindex="0">
```
[the remark this answers is not in the document]
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:22]: a scroll container must be keyboard accessible, it was an a11y error we got. with the tabindex, you can now tab to the table and use the arrow keys to scroll left/right top/bottom

https://github.com/SchwarzIT/onyx/pull/1200/commits/e45afdc5e12e66f7224b4de1064ac311d8d04b25
image

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 5]
```diff
@@ -105,7 +112,35 @@ const { densityClass } = useDensity(props);
   }
 }
 
+// todo try with grid...
```
[the remark this answers is not in the document]
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:115]: yep, missed it in cleanup, sry

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 6]
```diff
@@ -105,7 +112,35 @@ const { densityClass } = useDensity(props);
...
+
+    &__scroll-container {
+      border-radius: $border-radius;
+      overflow: auto;
+      box-sizing: border-box;
+      max-height: inherit;
+      max-width: inherit;
+      width: fit-content;
+    }
+    &__frame {
```
[the remark this answers is not in the document]
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:129]: did not work in a previous try but works with the current setup :) I'll update it

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue (hunk 3 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:159]: I'll talk with Jannick about the topic :)

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:22]: Alright :) But then we should propably also add styles for the outline like for the other components

[the remark this answers is not in the document]
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:22]: I'll ask jannick 👍

File: packages/sit-onyx/playwright/snapshots/components/OnyxTable/Table-scrolling--chromium-linux.png
Comment (larsrickert) [inline packages/sit-onyx/playwright/snapshots/components/OnyxTable/Table-scrolling--chromium-linux.png:1]: Should also include the row/column hover in the screenshot to be safe here?

Comment (BoppLi) [reply packages/sit-onyx/playwright/snapshots/components/OnyxTable/Table-scrolling--chromium-linux.png:1]: could I add this in the next PR for this ticket, once the "true" colors for the column hover are finalized?
I planned a meeting with UX tomorrow 10:00

**Review by larsrickert — APPROVED** [review]

**Pull request merged**